doctor/lib/text_extraction.py

```diff
@@ -0,0 +1,326 @@
+import re
+
+import pdfplumber
+from pdfplumber.ctm import CTM
+import pytesseract
+from pytesseract import Output
+import pandas as pd
+from PIL import Image
+
+
+def deskew(obj: dict) -> bool:
+    """Remove skewed text from a page
+
+    CTM stands for current transformation matrix.
+    Pdf plumber has a method to calculate the angle of text which we use here
+
+    Traditionally this is only seen in circular stamps which confuses the
+    content, or in perpendicular text of the ninth circuit courts which also
+    confuses the text.
+
+    :param obj: dictionary from pdfplumber for each word
+    :return: if the text should be returned
+    """
+    if (matrix := obj.get("matrix")) is None:
+        return True
+
+    # Remove Skew
+    my_char_ctm = CTM(*matrix)
+    if my_char_ctm.skew_x != 0:
+        return False
+    return True
+
+
+def get_page_text(page: pdfplumber.PDF.pages, strip_margin: bool) -> str:
+    """Extract page text
+
+    Using pdf plumber extract out the text of the document that is not
+    skewed (ie a stamp of approval) and extract out text removing blue text
+
+    :param page: PdfPlumber page
+    :param strip_margin: a flag to crop out the margin of a document and skewed content
+    :return: Text from the pdf plumber page
+    """
+    if strip_margin:
+        # Crop margins and remove skewed text
+        _, _, width, height = page.bbox
```

Comment:
See comment here: https://github.com/freelawproject/doctor/pull/187#discussion_r1602394162

Reply:
I dont understand why this matters?

Reply:
It doesn't matter much, but I had a hard time understanding your approach. I think I'd describe your version as "Margin percentage × width/height" and mine as "pixels per inch × location."

Reply:
thanks